docs/runtimes/node-affinity.md

```diff
@@ -1,96 +1,214 @@
+selector**, and adding or removing Key:Value pairs. Configure the node selection for individual Nuclio functions when creating a function in 
+the **Confguration** tab, under **Resources**, by adding Key:Value pairs.
+
+### SDK configuration
+
+Configure node selection by adding the key:value pairs in your Jupyter notebook formatted as a Python dictionary. <br>
+For example:
+
+```func.with_node_selection(node_selector={name})```
+
+See [with_node_selection](api/mlrun.runtimes.html?highlight=node_selector#mlrun.runtimes.RemoteRuntime.with_node_selection).
+
+
+
+## Volumes
+
+When you create a pod in an MLRun job or Nuclio function, the pod by default has access to a file-system which is ephemeral, and gets 
+deleted when the pod completes its execution. In many cases, a job requires access to files residing on external storage, or to files 
+containing configurations and secrets exposed through Kubernetes config-maps or secrets.
+Pods can be configured to consume the following types of volumes, and to mount them as local files in the local pod file-system:
+
+- V3IO containers: when running on the Iguazio system, pods have access to the underlying V3IO shared storage. This option mounts a V3IO container or a subpath within it to the pod through the V3IO FUSE driver.
+- PVC: Mount a Kubernetes persistent volume claim (PVC) to the pod. The persistent volume and the claim need to be configured beforehand.
+- Config Map: Mount a Kubernetes Config Map as local files to the pod.
+- Secret: Mount a Kubernetes secret as local files to the pod.
+
+For each of the options, a name needs to be assigned to the volume, as well as a local path to mount the volume at (using a Kubernetes Volume Mount). Depending on the type of the volume, other configuration options may be needed, such as an access-key needed for V3IO volume.
+
+See more about [Volumes](https://kubernetes.io/docs/concepts/storage/volumes/).
+
+MLRun supports the concept of volume auto-mount which automatically mounts the most commonly used type of volume to all pods, unless disabled. See more about [MLRun auto mount](../runtimes/function-storage.html).
+
+### UI configuration
+
+You can configure Volumes when creating a job, rerunning an existing job, and creating an ML function.
+Modify the Volumes for an ML function by pressing **ML functions**, then **<img src="../_static/images/kebab-menu.png" width="25"/>** 
+of the function, **Edit** | **Resources** | **Volumes** drop-down list. 
+
+Select the volume mount type: either Auto (using auto-mount), Manual or None. If selecting Manual, fill in the details in the volumes list .for each volume to mount to the pod. Multiple volumes can be configured for a single pod.
+
+### SDK configuration
+
+Configure Volumes attached to a function by using the `apply` function modifier on the function. For example:
+
+```func.apply(mlrun.platforms.mount_v3io())```
```

Comment:
need more elaborated example, explain the values/options, also for the other cases at least for PVC, and add hyperlinks to the py methods